mlir/include/mlir/Dialect/Rock/IR/MfmaInsnGroup.h

```diff
@@ -0,0 +1,124 @@
+//===- MfmaInsnGroup.h - MLIR to C++ for Rock conversion
+//---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// This file implements code selection logic for Mfma instructions.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_MFMA_INSN_GROUP_H
+#define MLIR_MFMA_INSN_GROUP_H
+
+#include <set>
+
+#include "mlir/Dialect/AMDGPU/AMDGPUDialect.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringMap.h"
+
+namespace mlir {
+namespace rock {
+
+enum class MfmaTypeId { Fp32TyId = 0, Fp16TyId, Bf16TyId, I8TyId };
+
+struct MfmaInsnInfo {
+  MfmaTypeId type;
+  int64_t mfmaNonKDim;
+  int64_t k;
+  int64_t blocksMfma;
+};
+
+struct MfmaInsnAttr {
```

Comment:
NIt: If this'll be called `attr`, can we perhaps have it generated by tablegen/be an actual attribute? Or alternatively can we go for a different name?

Reply:
Hmm so which would you recommend? I do plan on to make it carried with instruction in the lowering process in the next release. In that aspect I think attribute is accurate. So the short answer here is that I'll address it in the next PR (basically the points you have mentioned from your summary comments).

Reply:
Yeah, I'm not going to hold this up over naming complaints. Though `MfmaInsnInfo` is a good alternate name for this database if we want to not make it a `Attribute` subclass